rpki-validator/src/test/java/net/ripe/rpki/validator3/util/HappyEyeballsResolverTest.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * The BSD License
+ *
+ * Copyright (c) 2010-2018 RIPE NCC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   - Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *   - Redistributions in binary form must reproduce the above copyright notice,
+ *     this list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *   - Neither the name of the RIPE NCC nor the names of its contributors may be
+ *     used to endorse or promote products derived from this software without
+ *     specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.ripe.rpki.validator3.util;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class HappyEyeballsResolverTest {
+
+    private HappyEyeballsResolver subject = new HappyEyeballsResolver(new HttpClient());
+
+    @Test
+    public void should_recognise_ipv4_literal() {
+        assertThat(subject.isLiteralIpAddress("10.0.0.10")).hasValue("10.0.0.10");
+        assertThat(subject.isLiteralIpAddress("127.0.0.1")).hasValue("127.0.0.1");
+        assertThat(subject.isLiteralIpAddress("567.0.0.1")).hasValue("567.0.0.1"); // true in this implementation
```

Comment:
This is intentional, for 2 reasons.
I don't like to use Ipv4Address.parse because the only way for it to tell "it is not and IP address" is to throw an exception. This is too expensive, considering that in most cases the input won't be an IP address.
And I don't want to try to resolve a DNS name from something which is clearly not a DNS name. So doing "new SocketAddress()" and getting an exception from it is cheaper than doing a DNS call.